### ISSUE TITLE
Resolve unresolved vars when imported with potemkin

### ISSUE THREAD

**Issue by ericdallo** [clj-kondo/clj-kondo#1143]
**version**
`2021.01.20`

**platform**

JVM clj-kondo

**editor**

Emacs 28, happens with `clojure-lsp` and `clj-kondo` alone.

**problem**

clj-kondo warns about unresolved-var when the function is imported or is dynamically created.

**repro**
I create [this file](https://github.com/ericdallo/clojure-sample/blob/master/src/clojure_sample/issues/clj_kondo_unknown_var.clj) in a sample project to reproduce the issue.
When I run:
```bash
$ clj-kondo --lint  src/clojure_sample/issues/clj_kondo_unknown_var.clj

src/clojure_sample/issues/clj_kondo_unknown_var.clj:4:2: warning: Unresolved var: digest/md5
linting took 59ms, errors: 0, warnings: 1
```
this seems to happens because clj-kondo doesn't have info about the dynamic functions that are created in runtime:
```
$ cat .clj-kondo/.cache/2021.01.20/clj/digest.transit.json | lein jet --from transit --to json |jq 'keys'

[
  "*buffer-size*",
  "-digest",
  "Digestible",
  "algorithms",
  "byte-seq",
  "create-fn!",
  "create-fns",
  "digest",
  "read-some",
  "signature"
]
```
It seems to happen with `rewrite-clj.zip` too since the functions are imported from other namespaces: https://github.com/clj-commons/rewrite-clj/blob/master/src/rewrite_clj/zip.clj#L24-L79

**expected behavior**

Don't report as unresolved.

I think it's a corner case, I don't how many libs declare dynamic functions/import from other ns though

**Comment by borkdude** [comment]
Functions create dynamically aren't recognised by clj-kondo. This is not something that can be fixed by static analysis (easily). This is why clj-kondo supports configuration options. 

See https://github.com/clj-kondo/clj-kondo/blob/master/doc/linters.md#unresolved-var.

As for rewrite-clj.zip there might be something to fix, since clj-kondo has some logic around potemkin.

**Comment by ericdallo** [comment]
That's why I thought, great to hear about the exclude options :)
Thank you! Feel free to close the issue

**Comment by borkdude** [comment]
I renamed the issue to something I want to look into, so let's leave it open for now.

**Comment by ericdallo** [comment]
I found `clojure.core.async/alts!!` has the same issue because of a crazy odd macro:
https://github.com/clojure/core.async/blob/master/src/main/clojure/clojure/core/async.clj#L301-L312

**Comment by borkdude** [comment]
This can be solved using `:lint-as`.

**Comment by borkdude** [comment]
I checked the above problem with `rewrite-clj.potemkin`. Clj-kondo has built-in support for potemkin and when you lint-as potemkin it works as expected without any unresolved vars:

``` 
borkdude@MBP2019 /tmp/potemkin $ cat .clj-kondo/config.edn
{:lint-as {rewrite-clj.potemkin/import-vars potemkin/import-vars}}
borkdude@MBP2019 /tmp/potemkin $ cat src/foo.clj
(ns foo
  (:require [rewrite-clj.potemkin :refer [import-vars]]))

(import-vars
 [clojure.walk
  prewalk
  postwalk]
 [clojure.data
  diff])

borkdude@MBP2019 /tmp/potemkin $ cat src/bar.clj
(ns bar (:require [foo]))

(foo/diff [1] [2])

borkdude@MBP2019 /tmp/potemkin $ clj-kondo --lint src
linting took 13ms, errors: 0, warnings: 0
```
